Refactor deprecations, improve code style and drop owncloud support

st3iny opened the pull request:
Fixes #46

I took the liberty to fix the indentation in ChecksumController.php. Previously, tabs and spaces were mixed there.

As discussed in the linked issue, my changes will inevitably drop support for Owncloud. However, the usage of the modern Nextcloud API should make this app future proof for quite some time to come.

I strongly suggest to bump the version to 1.x.x after merging this PR.

cc @westberliner

westberliner commented on lib/Controller/ChecksumController.php:
```diff
@@ -1,82 +1,132 @@
+		$hash = hash_init($type);
+		hash_update_stream($hash, $file);
+		fclose($file);
+		return hash_final($hash);
```
Not sure if it is a psr-2 thing. Blank line before return.

westberliner commented on lib/Controller/ChecksumController.php:
```diff
@@ -1,82 +1,132 @@
+	}
+
+	private function getAllowedAlgorithmTypes(): array {
+		return array(
```
Short array syntax.

westberliner commented on lib/Listener/LoadAdditionalScriptsListener.php:
```diff
@@ -0,0 +1,40 @@
+use OCP\Util;
+
+class LoadAdditionalScriptsListener implements IEventListener {
+	public function handle(Event $event): void {
```
Blank line before.

westberliner reviewed (changes requested):
Thx for the contribution. Looks good so far.

I'll checked the [guidelines](https://docs.nextcloud.com/server/10.0/developer_manual/general/codingguidelines.html#coding) and if I am correct you are using tabs instead of spaces. Could you switch that? I found some minor cs stuff - I'll will marked them. If you do not have the time I will look into that in a way lots of these issues are mine anyway :)

I'll try to test it tomorrow and merge/fix it.

st3iny commented:
> I'll checked the guidelines and if I am correct you are using tabs instead of spaces. Could you switch that? 

@westberliner These guidelines state that one should use tabs for indentation (Quote: `Use tabs to indent`). But I'm fine with spaces too.

Thanks for your feedback. I implemented your suggestions and switched the indentation to spaces.

westberliner commented:
> @westberliner These guidelines state that one should use tabs for indentation (Quote: `Use tabs to indent`). But I'm fine with spaces too.

In the next line:

> A tab is 4 spaces wide.

But I think I misunderstood it. Most of the nextcloud code is tabs. Sorry about that. I will check your changes later that day. Thx so far.

westberliner commented:
Works :) Release is on the way.
